src/data_watcher.hpp

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <sys/inotify.h>
+#include <unistd.h>
+
+#include <sdbusplus/async.hpp>
+
+#include <filesystem>
+
+namespace data_sync
+{
+namespace watch::inotify
+{
+
+namespace fs = std::filesystem;
```

Comment:
I don't think this is used in this file?  I wouldn't put it here just for other files to use, so either remove it or move it to the .cpp if you want to use it there.